java-generator/gradle-plugin/src/main/java/io/fabric8/java/generator/gradle/plugin/JavaGeneratorPluginExtension.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.gradle.plugin;
+
+import io.fabric8.java.generator.Config;
+import org.gradle.api.Project;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.ListProperty;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public abstract class JavaGeneratorPluginExtension {
+
+  public static final String NAME = "javaGen";
+  private Project gradleProject;
+
+  private Config javaGeneratorConfig = new Config(
+      null,
+      null,
+      true,
+      null);
+
+  public JavaGeneratorPluginExtension(Project gradleProject) {
+    this.gradleProject = gradleProject;
+  }
+
+  public Config getConfig() {
+    return javaGeneratorConfig;
+  }
+
+  /**
+   * The input file or directory to be used for generating sources
+   *
+   */
+  public abstract RegularFileProperty getSource();
+
+  public File getSourceOrDefault() {
+    return this.getSource().getAsFile().getOrElse(null);
+  }
+
+  /**
+   * The URLs to be used to download CRDs from remote locations
+   */
+  public abstract ListProperty<String> getUrls();
+
+  public List<String> getUrlsOrDefault() {
+    return this.getUrls().getOrElse(new ArrayList<>());
+  }
+
+  /**
+   * The Download target folder for CRDs downloaded from remote URLs
+   *
+   */
+  public abstract DirectoryProperty getDownloadTarget();
+
+  public File getDownloadTargetOrDefault() {
+    return this.getDownloadTarget().getAsFile()
+        .getOrElse(this.gradleProject.getLayout().getProjectDirectory()
+            .dir("src")
```

Comment:
Usually, we download the CRD to a "temp" or "target" dir instead of a "managed source folder", something like:
`build/crds` wdyt?

Reply:
`build` would be in the Gradle lifecycle managment scope. If we want to use a default here - then either we live with a Java project structure and let the current structure be or we use something different, e.g.: `tmp/crds`, although this would mean for us to ger back to my initial implementation, basically.
TBH, we're generating Java sources, which to my eyes makes the current implementation legit, rather than the proposed one. I can be proved wrong, thoug, just let me know what you think.

Reply:
Now I got what you mean, i.e. putput the downloaded files to the build folder, so calling `clean` would... clean :slightly_smiling_face: 
Done, thanks!